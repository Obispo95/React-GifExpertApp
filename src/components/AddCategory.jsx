import { useState } from "react";
import {PropTypes} from 'prop-types'

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        
        event.preventDefault();

        //Evalua si tiene mas de un caracter, si no cumple, sale de la funcion
        if (inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
