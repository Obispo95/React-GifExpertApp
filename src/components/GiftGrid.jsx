import { GiftItem } from './GiftItem.jsx';
import { useFetchGifs } from '../hooks/useFetchGifs.js';


export const GiftGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Loading...</h2>)
            }

            <div className='card-grid'>
                {
                    images.map((image) => (
                        <GiftItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}
