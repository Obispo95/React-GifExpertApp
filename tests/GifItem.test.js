import { render, screen, fireEvent } from '@testing-library/react';
import { GifItem } from '../src/components/GifItem'

describe('Prueba para GiftItem', () => {

    const title = 'One Punch';
    const url = 'https://onepunch.com/image.jpg';

    test('Debe hacer match con snapshot ', () => {

        const { container } = render(<GifItem title={title} url={url} />)

        expect(container).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con url indicado', () => {

        render(<GifItem title={title} url={url} />)
        // screen.debug();

        expect(screen.getByRole('img').src).toBe(url);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(src);
        expect(url).toBe(url)
    })

    test('Debe de mostrar el titulo del componente', () => {

        render(<GifItem title={title} url={url} />)
        expect( screen.getByText(title) ).toBeTruthy();
    })
    
})