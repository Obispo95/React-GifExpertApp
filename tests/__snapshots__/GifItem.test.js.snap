// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para GiftItem Debe hacer match con snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://onepunch.com/image.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
